easybuild/easyblocks/j/java.py

```diff
@@ -27,13 +27,45 @@
 
 @author: Jens Timmerman (Ghent University)
 """
+
+import os
+import shutil
+
+from distutils.version import LooseVersion
 from easybuild.easyblocks.generic.packedbinary import PackedBinary
+from easybuild.tools.run import run_cmd
 
 
 class EB_Java(PackedBinary):
     """Support for installing Java as a packed binary file (.tar.gz)
     Use the PackedBinary easyblock and set some extra paths.
     """
+    
+    def extract_step(self):
+        """Unpack the source"""
+        if LooseVersion(self.version) < LooseVersion('1.7'):
+            try: 
+                os.chmod(self.src[0]['path'], 0755)
+            except OSError, err:
+                raise EasyBuildError("Failed adding execution permission to java installer: %s", err)
```

Comment:
@pescobar please use `adjust_permissions` (from `filetools`):

```python
adjust_permissions(self.src[0]['path'], stat.S_IXUSR, add=True)
```

Also, this will change permissions of the actual source file itself, should we worry about that (and copy it to build dir before changing permissions & running it)?